controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1018,6 +805,8 @@ void cli_bml::events_update_to_socket_cb(const struct BML_EVENT *event)
     cli_bml::events_update_cb(event, false);
 }
 
+void cli_bml::wfa_ca_reply_to_console_cb(const char *buffer) { std::cout << buffer << std::endl; }
```

Comment:
I see that the response is simple STDOUT. Why do you need all this complex callback and pointers mechanism? can't you just print the response once you get it?
It will simplify most of the logic you added to handle_response (pending_second, promises etc....)

Reply:
Because we need to handle two responses.
One response should be return at most in 1 second.
The second response may come after at most 120 seconds.  I can't block the cli between the two responses. So I must use callback which is similar to a response that is an "event" like.

Reply:
Reopening since we decieded to go with a different approach of handling everything in the callback.